featurebyte/service/feature_materialize_prerequisite.py

```diff
@@ -0,0 +1,140 @@
+        raise DocumentNotFoundError("FeatureMaterializePrerequisite document not found")
+
+    async def get_document_for_feature_table(
+        self,
+        offline_store_feature_table_id: ObjectId,
+        scheduled_job_ts: datetime,
+    ) -> FeatureMaterializePrerequisite:
+        """
+        Get the FeatureMaterializePrerequisite document for the feature table and job time
+
+        Parameters
+        ----------
+        offline_store_feature_table_id: ObjectId
+            Offline store feature table identifier
+        scheduled_job_ts: datetime
+            Scheduled job time used to identify the current job cycle
+
+        Returns
+        -------
+        FeatureMaterializePrerequisite
+
+        Raises
+        ------
+        DocumentNotFoundError
+            If the document to be updated does not exist
+        """
+        async for model in self.list_documents_iterator(
+            query_filter={
+                "offline_store_feature_table_id": offline_store_feature_table_id,
+                "scheduled_job_ts": scheduled_job_ts,
+            },
+        ):
+            return model
+        raise DocumentNotFoundError("FeatureMaterializePrerequisite document not found")
+
+    async def add_completed_prerequisite(
+        self,
+        offline_store_feature_table_id: ObjectId,
+        scheduled_job_ts: datetime,
+        prerequisite_tile_task: PrerequisiteTileTask,
+    ) -> None:
+        """
+        Insert a completed task item into an existing prerequisite document
+
+        Parameters
+        ----------
+        offline_store_feature_table_id: ObjectId
+            Offline store feature table identifier
+        scheduled_job_ts: datetime
+            Scheduled job time used to identify the current job cycle
+        prerequisite_tile_task: PrerequisiteTileTask
+            Representation of a completed tile task
+        """
+        try:
+            document_id = await self.get_document_id_for_feature_table(
+                offline_store_feature_table_id, scheduled_job_ts
+            )
+        except DocumentNotFoundError:
+            # Handle the edge case where the prerequisite document is not yet created. This should
+            # be rare since it should be much faster to create this document than completing a tile
+            # task.
+            feature_materialize_prerequisite = FeatureMaterializePrerequisite(
+                offline_store_feature_table_id=offline_store_feature_table_id,
+                scheduled_job_ts=scheduled_job_ts,
+            )
+            document_id = (await self.create_document(feature_materialize_prerequisite)).id
+        query_filter = self._construct_get_query_filter(document_id)
+        await self.persistent.update_one(
+            collection_name=self.collection_name,
+            query_filter=query_filter,
+            update={"$push": {"completed": prerequisite_tile_task.dict(by_alias=True)}},
+            user_id=self.user.id,
+        )
```

Comment:
Could consider to use `self.update_documents` here.

Reply:
Indeed, with that we only need to specify query filter and update schema and the rest are automatically set. Will switch to that in a follow up PR.